### ISSUE TITLE
Fatal error: Uncaught ReflectionException: Class "config" does not exist

### ISSUE THREAD

**Issue by adiafora** [spatie/laravel-permission#2097]
If I want to adhere to the principle of dependency injection, and add `Spatie\Permission\Models\Role` to the __construct of any class, I will get an error `Fatal error: Uncaught ReflectionException: Class "config" does not exist`

Because the helper `config()` is not initialized yet at the moment.

A simple example:

```php

namespace App\Services\Permission;

class PermissionService
{
    public function __construct(
        private \Spatie\Permission\Models\Role $role,
    ) {
    }
}
```

Now in the terminal, try typing a command, for example `php artisan cache:clear`, and you will get an error.
I have a solution, I will offer it in PR ))

**Versions**
laravel/framework -  v8.83.11
spatie/laravel-permission - 5.5.3 
PHP 8.0.8 
Windows 11

**Comment by adiafora** [comment]
@systemsolutionweb Yes, I looked more closely and found that only using dependency injection in the code does not give such an error. But if you use this `PermissionService` class in DI in the `App\Console\Kernel` constructor, then there will be an error with any access to the terminal.

```php
namespace App\Console;

use Illuminate\Foundation\Console\Kernel as ConsoleKernel;

class Kernel extends ConsoleKernel
{
    public function __construct(
        Application $app,
        Dispatcher $events,
        private \App\Services\Permission $permission;
    }
}
```

```php
namespace App\Services\Permission;

class PermissionService
{
    public function __construct(
        private \Spatie\Permission\Models\Role $role,
    ) {
    }
}
```

**Comment by systemsolutionweb** [comment]
Looks like laravel bug, the behavior must be the same on both cases

**Comment by systemsolutionweb** [comment]
Looks like laravel bug, the behavior must be the same on both cases
Also, on kernel __construct configurations not loaded yet
`\Illuminate\Foundation\Bootstrap\LoadConfiguration::class`
https://github.com/laravel/framework/blob/9da642c4f7fa5d294551835ca6731e7f807e2125/src/Illuminate/Foundation/Console/Kernel.php#L62-L70
https://github.com/laravel/framework/blob/9da642c4f7fa5d294551835ca6731e7f807e2125/src/Illuminate/Foundation/Console/Kernel.php#L307-L320
